index.js

```diff
@@ -62,18 +62,48 @@ function gql(/* arguments */) {
   args.shift();
   var substitutions = args;
 
+  var fragments = []
   var result = '';
 
   // run the loop only for the substitution count
   for (var i = 0; i < substitutions.length; i++) {
-      result += literals[i];
+    var substitution = substitutions[i]
+    var substitutionType = typeof substitution
+
+    result += literals[i];
+
+    if (substitutionType === 'string') {
+      console.warn('gql: Interpolation of arbitrary strings is deprecated')
       result += substitutions[i];
+    } else if (
+      substitutionType === 'object' && substitution.kind === 'Document'
+    ) {
+      var fragment = substitution.definitions[0]
```

Comment:
Documents can have multiple fragments, so we should either support that or error when there is more than one fragment.